src/measure_theory/conditional_expectation.lean

```diff
@@ -0,0 +1,267 @@
+by { obtain ⟨g, hg_meas, hfg⟩ := hf, exact ⟨g, hg_meas, ae_eq_of_ae_eq_trim hfg⟩, }
+
+variables {α β γ E E' F F' G G' H 𝕜 : Type*} {p : ℝ≥0∞}
+  [is_R_or_C 𝕜] [measurable_space 𝕜] -- 𝕜 for ℝ or ℂ, together with a measurable_space
+  [measurable_space β] -- β for a generic measurable space
+  -- E and E' will be used for inner product spaces, when they are needed.
+  -- F for a Lp submodule
+  [normed_group F] [normed_space 𝕜 F] [measurable_space F] [borel_space F]
+  [second_countable_topology F]
+  -- F' for integrals on a Lp submodule
+  [normed_group F'] [normed_space 𝕜 F'] [measurable_space F'] [borel_space F']
+  [second_countable_topology F'] [normed_space ℝ F'] [complete_space F']
+  -- G for a Lp add_subgroup
+  [normed_group G] [measurable_space G] [borel_space G] [second_countable_topology G]
+  -- G' for integrals on a Lp add_subgroup
+  [normed_group G'] [measurable_space G'] [borel_space G'] [second_countable_topology G']
+  [normed_space ℝ G'] [complete_space G']
+  -- H for measurable space and normed group (hypotheses of mem_ℒp)
+  [measurable_space H] [normed_group H]
+
+section Lp_meas
+
+variables (F 𝕜)
+/-- `Lp_meas F 𝕜 m p μ` is the subspace of `Lp F p μ` containing functions `f` verifying
+`ae_measurable' m f μ`, i.e. functions which are `μ`-a.e. equal to an `m`-measurable function. -/
+def Lp_meas [opens_measurable_space 𝕜] (m : measurable_space α) [measurable_space α] (p : ℝ≥0∞)
+  (μ : measure α) :
+  submodule 𝕜 (Lp F p μ) :=
+{ carrier   := {f : (Lp F p μ) | ae_measurable' m f μ} ,
+  zero_mem' := ⟨(0 : α → F), @measurable_zero _ α _ m _, Lp.coe_fn_zero _ _ _⟩,
+  add_mem'  := λ f g hf hg, (hf.add hg).congr (Lp.coe_fn_add f g).symm,
+  smul_mem' := λ c f hf, (hf.const_smul c).congr (Lp.coe_fn_smul c f).symm, }
+variables {F 𝕜}
+
+variables [opens_measurable_space 𝕜]
+
+lemma mem_Lp_meas_iff_ae_measurable' {m m0 : measurable_space α} {μ : measure α} {f : Lp F p μ} :
+  f ∈ Lp_meas F 𝕜 m p μ ↔ ae_measurable' m f μ :=
+by simp_rw [← set_like.mem_coe, ← submodule.mem_carrier, Lp_meas, set.mem_set_of_eq]
+
+lemma Lp_meas.ae_measurable' {m m0 : measurable_space α} {μ : measure α} (f : Lp_meas F 𝕜 m p μ) :
+  ae_measurable' m f μ :=
+mem_Lp_meas_iff_ae_measurable'.mp f.mem
+
+lemma mem_Lp_meas_self {m0 : measurable_space α} (μ : measure α) (f : Lp F p μ) :
+  f ∈ Lp_meas F 𝕜 m0 p μ :=
+mem_Lp_meas_iff_ae_measurable'.mpr (Lp.ae_measurable f)
+
+lemma Lp_meas_coe {m m0 : measurable_space α} {μ : measure α} {f : Lp_meas F 𝕜 m p μ} :
+  ⇑f = (f : Lp F p μ) :=
+coe_fn_coe_base f
+
+section complete_subspace
+
+/-! ## The subspace `Lp_meas` is complete.
+
+We define a `linear_isometry_equiv` between `Lp_meas` and the `Lp` space corresponding to the
+measure `μ.trim hm`. As a consequence, the completeness of `Lp` implies completeness of
+`Lp_meas`. -/
+
+variables {ι : Type*} {m m0 : measurable_space α} {μ : measure α}
+
+lemma mem_ℒp_trim_of_mem_Lp_meas (hm : m ≤ m0) (f : Lp F p μ) (hf_meas : f ∈ Lp_meas F 𝕜 m p μ) :
```

Comment:
can you add docstrings to this lemma and the next one? The presence of @ and of proof arguments in the statements make them a little bit hard to parse (but I agree that this is unavoidable!)